src/inferencerequest.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+//*****************************************************************************
+// Copyright 2022 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <string>
+#include <unordered_map>
+
+#include "inferenceparameter.hpp"
+#include "inferencetensor.hpp"
+#include "pocapi.hpp"
+
+namespace ovms {
+
+class Status;
+
+class InferenceRequest {
+    const std::string servableName;
+    const uint64_t servableVersion;
+    std::unordered_map<std::string, InferenceParameter> parameters;
+    std::unordered_map<std::string, InferenceTensor> inputs;
+
+public:
+    InferenceRequest(const char* modelName, uint64_t modelVersion);
+    Status addInput(const char* name, DataType datatype, const size_t* shape, size_t dimCount);
```

Comment:
Why use size_t* here if we can use classes/structs in our cpp impl?

Reply:
It's a matter of choice but in most cases pure casting to array of size_t is the easiest to use IMO.